src/python_testing/mdns_helper.py

```diff
@@ -0,0 +1,159 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from mobly import asserts
+from zeroconf import ServiceListener, Zeroconf
+from zeroconf.asyncio import AsyncServiceInfo, AsyncZeroconf, AsyncZeroconfServiceTypes
+
+MDNS_TYPE_OPERATIONAL = "_matter._tcp.local."
+
+'''
+A service listener for the TXT record data to get populated
+'''
+
+
+class DummyServiceListener(ServiceListener):
+
+    def add_service(self, zc: Zeroconf, type: str, name: str) -> None:
+        pass
+
+    def remove_service(self, zc: Zeroconf, type: str, name: str) -> None:
+        pass
+
+    def update_service(self, zc: Zeroconf, type: str, name: str) -> None:
+        pass
+
+
+class MdnsHelper:
+    def __init__(self, dut):
+        self._listener = DummyServiceListener()
+        self._azc: AsyncZeroconf = AsyncZeroconf()
+        self._dut = dut
+        self._service_types = None
+        self._service_info = None
+        self._name = f"{self._get_service_name()}.{MDNS_TYPE_OPERATIONAL}"
+
+    def _get_service_name(self):
+        node_id = self._dut.dut_node_id
+        node_id_hex = str(hex(int(node_id))[2:].upper()).zfill(16)
+        compressed_fabricid = self._dut.default_controller.GetCompressedFabricId()
+        compressed_fabricid_hex = hex(int(compressed_fabricid))[2:].upper()
+        service_name = f"{compressed_fabricid_hex}-{node_id_hex}"
+        return service_name
+
+    def _get_txt_record_key_value(self, key: str):
+        # Convert the key from string to bytes, as the dictionary uses bytes
+        byte_key = key.encode('utf-8')
+
+        # Check if the key exists in the dictionary
+        asserts.assert_in(byte_key, self._service_info.properties, f"Property '{key}' not found")
+        value = self._service_info.properties[byte_key]
+
+        # Check if the value is not None
+        asserts.assert_is_not_none(value, f"Value for property '{key}' not present")
```

Comment:
Can you please move this external to the class? That will make this class more generally useful for other tests in the future.